modules/end-user-guide/pages/adding-a-vscode-extension.adoc

```diff
@@ -0,0 +1,163 @@
+
+[id="plug-in-registries"]
+== {prod-short} plug-in registries overview
+
+{prod-short} has a default plug-in and extensions registry installed in every {prod-short} instance. The Che-Theia IDE searches and installs the plug-ins and extensions from that location.
+
+Check this {prod-short} https://github.com/eclipse-che/che-plugin-registry[registry project] for an overview of default plug-ins, extensions, and source code. An online instance that is refreshed after every commit to the main branch, is located https://eclipse-che.github.io/che-plugin-registry/main/v3/plugins/[here]. You can use a different plug-in or extension registry for Che-Theia if you don't work in air gap mode: only the default registry is available there.
+
+The plug-in and extension overview for Che-Code Visual Studio Code editor is located in the https://www.open-vsx.org/[OpenVSX instance]. Air gap mode is not yet supported for this editor.
+
+[id="visual-studio-code-extensions-json"]
+== Adding an extension to `.vscode/extensions.json`
+
+The easiest way to add a Visual Studio Code extension to a workspace is to add it to the `.vscode/extensions.json` file. The main advantage of this method is that it works with all supported {prod-short} IDEs.
+
+If you use the Che-Theia IDE, the extension is installed and configured automatically. If you use a different supported IDE with the Che-Code Visual Studio Code fork, a pop-up with a recommendation to install the extension shows when you start the workspace.
+
+.Prerequisites
+. You have the `.vscode/extensions.json` file in the root of the GitHub repository.
+
+.Procedure
+. Add the extension id to `.vscode/extensions.json` file. Use a `.` sign to separate the publisher and extension names:
++
+*EXAMPLE*
+[source,json,subs="+quotes"]
+----
+  {
+    "recommendations": [
+      "redhat.java"
+    ]
+  }
+----
+
+*NOTE* 
+If the specified set of extension ids isn't available in the {prod-short} registry, the workspace starts without the extension.
+
+
+[id="che-theia-plug-ins-YAML"]
+== Adding plug-in parameters to `.che/che-theia-plugins.yaml`
+You can add extra parameters to a plug-in by modifying the `.che/che-theia-plugins.yaml` file. These modifications include:
+* Defining the plug-ins for workspace installation.
+* Changing default memory limit.
+* Overriding default preferences.
+
+
+=== Defining the plug-ins for workspace installation
+Define the plug-ins to be installed in the workspace.
+
+.Prerequisites
+. You have the `.che/che-theia-plugins.yaml` file in the root of the GitHub repository.
+
+.Procedure
+. Add the id of the plug-in in the  `.che/che-theia-plugins.yaml` file. Use the `/` sign to separate the publisher and plug-in name:
++
+*EXAMPLE*
+[source,yaml,subs="+quotes"]
+----
+- id: redhat/java  
+----
+
+=== Changing default memory limit
+
+Override container settings such as the memory limit.
```

Comment:
It sounds like this procedure might  demonstrate how to override other settings besides the memory limit, but then it doesn't mention what they are. Maybe make this phrase more precise or mention the other settings below?